### PR TITLE
plan: fix bug in prepare statement.

### PR DESCRIPTION
**Pull request by winoros**
When doing prepare and execute statement, the range of index scan is not set correctly.
This previous unit-test doesn't open the prepare statement cache in fact.

### CODE REVIEW

File: plan/common_plans.go [hunk 1]
```diff
@@ -256,7 +256,7 @@ func (e *Execute) buildRangeForIndexScan(sctx sessionctx.Context, is *PhysicalIn
 	ranges := ranger.FullRange()
 	if len(idxCols) > 0 {
 		var err error
-		ranges, _, _, _, err = ranger.DetachCondAndBuildRangeForIndex(sctx, is.conditions, idxCols, colLengths)
+		ranges, _, _, _, err = ranger.DetachCondAndBuildRangeForIndex(sctx, is.AccessCondition, idxCols, colLengths)
```
Comment (zz-jason) [inline plan/common_plans.go]: Can we remove field `conditions` from `PhysicalIndexScan`?

File: executor/prepared_test.go [hunk 2]
```diff
@@ -25,14 +24,13 @@ import (
...
-	orgEnable := cfg.PreparedPlanCache.Enabled
-	orgCapacity := cfg.PreparedPlanCache.Capacity
+	orgEnable := plan.PreparedPlanCacheEnabled
+	orgCapacity := plan.PreparedPlanCacheCapacity
 	flags := []bool{false, true}
 	ctx := context.Background()
 	for _, flag := range flags {
-		cfg.PreparedPlanCache.Enabled = flag
-		cfg.PreparedPlanCache.Capacity = 100
+		plan.PreparedPlanCacheEnabled = flag
```
Comment (zz-jason) [inline executor/prepared_test.go]: Will it cause data race during test? Maybe other goroutine is changing this global variable concurrently during test.

Comment (winoros) [reply executor/prepared_test.go]: Seem `cfg` and `plan` are all global variable? So it's safe to do this?

File: plan/common_plans.go (hunk 1 above)
Comment (winoros) [reply plan/common_plans.go]: Done.

**Comment by winoros** [comment]
PTAL @XuHuaiyu @zz-jason

**Comment by zz-jason** [comment]
@coocood @XuHuaiyu PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**